src/core/etl/tests/Flow/ETL/Tests/Unit/DataFrameTest.php

```diff
@@ -1714,9 +1714,9 @@ public function test_strict_validation_against_schema() : void
             )
         )->validate(
             new Schema(
-                Schema\Definition::integer('id', $nullable = false),
-                Schema\Definition::string('name', $nullable = true),
-                Schema\Definition::boolean('active', $nullable = false),
+                Schema\Definition::integer('id', ScalarType::integer32()),
```

Comment:
Yeah, we need to decide here if we prefer to use `\Definition::integer()` or `\Definition::integer32()` - my problem with passing Type as a second parameter is that we need to validate it, which pretty much defeats the purpose of strongly typed arguments.
Alternatively, we can just drop all static factory methods (string/integer/boolean) and just use the constructor, however, that looks a bit less user-friendly. 

Additionally, I would put type into all Definition types, String, Boolean, Integer, we can also unify the metadata name to be something like: 

`METADATA_PHP_TYPE` which would always hold the instance of Type.

Reply:
I was not sure either how to decide about default that's why it's detecting inside what's max, but that's "magic", so I would go with BC break and force to choose either 32 or 64 on the user side but internally use 64 (similarly as it's now ie in parquet schema converter), with that I would also add check and throw exception when 64 is used when available is max 32.

So it would be: `Entry::integer64()`/`Entry::integer32()`

Other than that I would save the helpers as DX is good there. Type for scalar is saved too.

Reply:
I was thinking about it over night, and I can't figure out any good use case for the end users to decide if they want to use int64 vs int63, yeah, one requires less space but with current storage prices I'm not sure how relevant that is. 
Originally I was planning to use that in order to know how to map Flow Int into Parquet Int and maybe we just keep it this way. Let me think more about it, I'm closing this PR for now, will get back to it